runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcWindmillServer.java

```diff
@@ -242,17 +239,17 @@ private synchronized void initializeLocalHost(int port) {
     Channel channel = localhostChannel(port);
     if (options.isEnableStreamingEngine()) {
       this.stubList.add(CloudWindmillServiceV1Alpha1Grpc.newStub(channel));
-      this.syncStubList.add(CloudWindmillServiceV1Alpha1Grpc.newBlockingStub(channel));
     } else {
-      this.syncApplianceStub = WindmillApplianceGrpc.newBlockingStub(channel);
+      this.syncApplianceStub =
+          WindmillApplianceGrpc.newBlockingStub(channel)
+              .withDeadlineAfter(unaryDeadlineSeconds, TimeUnit.SECONDS);
```

Comment:
can unaryDeadlineSeconds be removed if the stub remembers it now?

Reply:
done.

Reply:
done.